cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,138 @@
+		// 1. Set identity on Relay Chain
+		assert_ok!(<RococoRelay as RococoRelayPallet>::Identity::set_identity(
+			rococo_runtime::RuntimeOrigin::signed(RococoRelaySender::get()),
+			Box::new(identity_relaychain)
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Identity(pallet_identity::Event::IdentitySet { .. }) => {},
+			]
+		);
+
+		// 2. Set sub-identity on Relay Chain
+		assert_ok!(<RococoRelay as RococoRelayPallet>::Identity::set_subs(
+			rococo_runtime::RuntimeOrigin::signed(RococoRelaySender::get()),
+			vec![(RococoRelaySender::get(), Data::Raw(vec![40; 1].try_into().unwrap()))],
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Reserved { .. }) => {},
+				RuntimeEvent::Identity(pallet_identity::Event::IdentitySet { .. }) => {},
+				RuntimeEvent::Balances(pallet_balances::Event::Reserved { .. }) => {},
+			]
+		);
+
+		bal_after_relaychain =
+			<RococoRelay as RococoRelayPallet>::Balances::free_balance(RococoRelaySender::get());
+	});
+
+	// Set identity and Subs on Parachain with Zero deposit
+	PeopleRococo::execute_with(|| {
+		type RuntimeEvent = <PeopleRococo as Chain>::RuntimeEvent;
+
+		bal_before_parachain =
+			<PeopleRococo as PeopleRococoPallet>::Balances::free_balance(PeopleRococoSender::get());
+
+		// 3. Set identity on Parachain with zero deposit
+		assert_ok!(<PeopleRococo as PeopleRococoPallet>::Identity::set_identity_no_deposit(
+			&PeopleRococoSender::get(),
+			identity_parachain.clone()
+		));
+
+		// 4. Set sub-identity on Parachain
+		assert_ok!(<PeopleRococo as PeopleRococoPallet>::Identity::set_subs(
+			people_rococo_runtime::RuntimeOrigin::signed(PeopleRococoSender::get()),
+			vec![(PeopleRococoSender::get(), Data::Raw(vec![40; 1].try_into().unwrap()))],
+		));
+		assert_expected_events!(
+			PeopleRococo,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Reserved { .. }) => {},
+			]
+		);
+
+		bal_after_parachain =
+			<PeopleRococo as PeopleRococoPallet>::Balances::free_balance(PeopleRococoSender::get());
+		let fees = bal_before_parachain - bal_after_parachain;
+		assert_eq!(bal_before_parachain - fees, bal_after_parachain);
+	});
+
+	// 5. reap_identity on Relay Chain
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+		assert_ok!(<RococoRelay as RococoRelayPallet>::Identity::reap_identity(
+			&RococoRelaySender::get(),
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Unreserved { .. }) => {},
+			]
+		);
+		assert!(<RococoRelay as RococoRelayPallet>::Identity::identity(&RococoRelaySender::get())
+			.is_none());
+		assert!(<RococoRelay as RococoRelayPallet>::Identity::super_of(&RococoRelaySender::get())
+			.is_none());
+		let bal_after_reap =
+			<RococoRelay as RococoRelayPallet>::Balances::free_balance(RococoRelaySender::get());
+		assert_eq!(bal_after_reap, bal_before_relaychain)
```

Comment:
This assertion is wrong, some of the balance should have been teleported to the parachain. This should also be checked.